src/utils/helpers.ts

```diff
@@ -110,6 +110,24 @@ export async function getManifest(
 	const init = await import('balena-device-init');
 	const sdk = getBalenaSdk();
 	const manifest = await init.getImageManifest(image);
+	if (manifest != null) {
+		const config = manifest.configuration?.config;
+		if (config?.partition != null) {
+			const { getBootPartition } = await import('balena-config-json');
+			// Overwrite the deprecated  & no longer updated partition number on the device-type.json
```

Comment:
It seems odd to sometimes update config.partition vs. config.partition.primary with the same information. Is config.partition.primary going away?

Reply:
@kb2ma afaict both `partition: number` & `partition.primary: number` can be found in `/device-types/v1`. I have no clue whether anyone of these is going away. My intention is to check which of the two forms is used in the current DT and update it.
I updated the comment to make that a bit clearer.